realm/realm-library/src/androidTestObjectServer/java/io/realm/SchemaTests.java

```diff
@@ -145,4 +147,54 @@ public void allow_addField() {
 
         realm.close();
     }
+
+    @Test
+    public void addPrimaryKey_notAllowed() {
+        String className = "StringOnly";
+        Realm realm = Realm.getInstance(config);
+
+        realm.beginTransaction();
+        RealmObjectSchema objectSchema = realm.getSchema().get(className);
+        objectSchema.addField("foo", String.class);
+
+        try {
+            objectSchema.addPrimaryKey("foo");
```

Comment:
If we are releasing this as part of a 4.0, why not just remove this method completely?